docs/dev_guide/code_coverage_guide.md

```diff
@@ -0,0 +1,105 @@
+---
+layout: default
+title: Code Coverage Guide
+parent: Developer Guides
+permalink: /docs/dev_guide/code_coverage_guide
+---
+# Code Coverage Guide
+
+The Venice repository has two code coverage check on GitHub Action, submodule-level coverage verification
+and new-commit coverage verification, which aim to improve the testability and code quality overall. The
+GitHub Action fails the Pull Request if the coverage check doesn't meet the requirements. This guide provides the details
+on how these reports are generated and how to debug them.
+
+## Module structure
+
+Venice has 4 modules, `all-modules`, `clients`, `internal` and `services`. Each module has its own submodules, except `all-modules`.
+
+#### Clients
+
+`da-vinci-client`, `venice-admin-tool`, `venice-client`, `venice-push-job`, `venice-samza`, `venice-thin-client`
```

Comment:
Should we link to the directory on GH itself rather than enumerate here? If we change the module structure, I feel it's likely that we'll forget to update this, and it does not seem critical to the reader to have the list enumerated in line. Just some minor thought, but IDK.

Reply:
My original intention is to let reader know what we mean by `module` and `submodule` so they can find the reports under the right directory. Showing the example is the most straightforward way of doing it. I think we can leave it here and people will get the gist? Maybe I can add a disclaimer saying the structure might change over the time.